### DISCUSSION TITLE
Which engine does MiniDexed use?

### DISCUSSION THREAD

**Discussion by donluca** [probonopd/MiniDexed#439] [Q&A] [answered]
Dexed lets the user choose among 3 possible engines, taken form the homepage:

> Dexed comes with 3 engine types :
>
> Modern : this is the original 24-bit music-synthesizer-for-android implementation.
> Mark I : Based on the OPL Series but at a higher resolution (LUT are 10-bits). The target of this engine is to be > closest to the real DX7.
> OPL Series : this is an experimental implementation of the [reversed engineered OPL family chips](http://sourceforge.net/projects/peepeeplayer). 8-bit. Keep in mind that the envelopes stills needs tuning.

Mark I is the engine used by default (and the most accurate to an original DX7) on newer Dexed versions.

Which engine does MiniDexed use?

Also, I don't know if there is one engine which is more lightweight on the CPU, but, if there is one, letting the user choose a less resource intensive engine could open more possibilities on the lower end Raspberry Pi like Pi1 and Pi Zero.

**Comment by probonopd** [comment, marked as answer]
As per https://github.com/probonopd/MiniDexed/discussions/358, _Modern_ is used.

**Comment by probonopd** [comment]
Thanks to @dcoredump and @donluca we have a test build with the MKI ("[Mark I](https://en.wikipedia.org/wiki/Yamaha_DX7)"), and OPL ("[OPL Series](https://en.wikipedia.org/wiki/Yamaha_OPL)") engines in addition to the default one. Please feel free to test and leave your feedback there:

https://github.com/probonopd/MiniDexed/pull/443#issuecomment-1445444488

**Reply by Banana71** [reply]
Many thanks to @donluca, @dcoredump  and @probonopd. I'm looking forward to testing the engiens. Whether my old ears can hear the differences?

**Reply by donluca** [reply]
IIRC there are some instruments/filters which are just plain wrong with Modern (and there are still some which still need work on Mark I), but the best and quickest way to tell is to download the standalone [Dexed](https://asb2m10.github.io/dexed/) and try to switch back and forth from Modern to Mark I with the same instruments and hear it for yourself.

I swear there was a Youtube video I saw months ago comparing the engine, but I can't find it anymore on my browser's history.
